crates/core/src/models/transaction_receipt.rs

```diff
@@ -0,0 +1,136 @@
+
+pub struct StarknetTransactionReceipt(MaybePendingTransactionReceipt);
+
+impl From<MaybePendingTransactionReceipt> for StarknetTransactionReceipt {
+    fn from(receipt: MaybePendingTransactionReceipt) -> Self {
+        Self(receipt)
+    }
+}
+
+impl From<StarknetTransactionReceipt> for MaybePendingTransactionReceipt {
+    fn from(receipt: StarknetTransactionReceipt) -> Self {
+        receipt.0
+    }
+}
+
+#[async_trait]
+impl ConvertibleStarknetTransactionReceipt for StarknetTransactionReceipt {
+    async fn to_eth_transaction_receipt<P: Provider + Send + Sync>(
+        self,
+        client: &KakarotClient<P>,
+    ) -> Result<Option<EthTransactionReceipt>, EthApiError<P::Error>> {
+        let starknet_tx_receipt: MaybePendingTransactionReceipt = self.into();
+
+        let res_receipt = match starknet_tx_receipt {
+            MaybePendingTransactionReceipt::Receipt(receipt) => match receipt {
+                TransactionReceipt::Invoke(InvokeTransactionReceipt {
+                    transaction_hash,
+                    status,
+                    block_hash,
+                    block_number,
+                    events,
+                    ..
+                }) => {
+                    let starknet_tx: StarknetTransaction =
+                        client.starknet_provider().get_transaction_by_hash(transaction_hash).await?.into();
+
+                    let transaction_hash: Felt252Wrapper = transaction_hash.into();
+                    let transaction_hash: Option<H256> = Some(transaction_hash.into());
+
+                    let block_hash: Felt252Wrapper = block_hash.into();
+                    let block_hash: Option<H256> = Some(block_hash.into());
+
+                    let block_number: Felt252Wrapper = block_number.into();
+                    let block_number: Option<U256> = Some(block_number.into());
+
+                    let eth_tx = starknet_tx.to_eth_transaction(client, None, None, None).await?;
+                    let from = eth_tx.from;
+                    let to = eth_tx.to;
+                    let contract_address = match to {
+                        // If to is Some, means contract_address should be None as it is a normal transaction
+                        Some(_) => None,
+                        // If to is None, is a contract creation transaction so contract_address should be Some
+                        None => {
+                            let event = events
+                                .iter()
+                                .find(|event| event.keys.iter().any(|key| *key == EVM_CONTRACT_DEPLOYED))
+                                .ok_or(EthApiError::Other(anyhow::anyhow!(
+                                    "Kakarot Core: No contract deployment event found in Kakarot transaction receipt"
+                                )))?;
+
+                            let evm_address =
+                                event.data.first().ok_or(DataDecodingError::InvalidReturnArrayLength {
```

Comment:
has this become the last element as per the last kakarot PR?

Reply:
It has become the first element so this should be ok